blog/posts/2021-02-21-serverless-functions-ember-glimmer.md

```diff
@@ -0,0 +1,213 @@
+
+```toml
+  [[redirects]]
+    from = "/ember-api-routes/*"
+    to = "/.netlify/functions/:splat"
+    status = 200
+```
+
+Now if we visit `http://localhost:8888/ember-api-routes/hello-world` in our browser we should see:
+
+```js
+{
+  message: "Hello World"
+}
+```
+
+Boom, we've got API routes on Ember working locally using Netlify Dev 🔥! 
+
+
+Make a few tweaks, add an `application.js` route that uses fetch to get data from the new function at `/ember-api-routes/hello-world`, and you can use it just as you would most any API. Here we're returning the response to the `application.js` route model hook and using it in the `application.hbs` template, passing it down as an argument to Ember's `<WelcomePage />` component to display the message.
+
+![Hello World Serverless Function Ember](/img/blog/serverless-functions-ember-glimmer/ember-hello-world.png)
+
+Deploying to Netlify is also super easy, but we won't cover that subject here as there are plenty of resources around it. Suffice it to say that if you deploy your site with Netlify it should JustWork™️ in the same way you just experienced in your local dev environment. Netlify will handle deployment of the serverless functions seamlessly.
+
+There are a ton of different ways we can use these serverless functions in our project, so use them as you see fit.
+
+### Netlify functions with Glimmer
+
+Okay, but what about when your framework is not part of the magical framework detection provided by Netlify Dev? Fortunately, that only requires a little bit of additional setup that we'll cover next.
+
+Follow the same steps described above, but with your Glimmer project. I may note that I used [glimmerx](https://github.com/glimmerjs/glimmer-experimental/) for purposes of this investigation. YMMV if you're using a different version of glimmer.
+
+You will notice that because there is no framework detected that Netlify Dev supports you won't have preset defaults to help you through the cli wizard after running `netlify init`. 
+
+No problem. Here is the output from the cli and the answers you'll need to get up and running:
+
+```bash
+? Your build command (hugo build/yarn run build/etc): npm run build
+? Directory to deploy (blank for current dir): dist
+? Netlify functions folder: functions
+? No netlify.toml detected. Would you like to create one with these build settings? Yes
+```
+
+Create your function in the same way that we did for Ember using `netlify functions:create hello-world`.
+
+Now go ahead and open the `netlify.toml` file that was just autogenerated as we'll need to add a few things to this. First, Netlify Dev needs to know more about your framework since it was not detected. Go ahead and add this to the toml:
```

Comment:
```suggestion
Now let's open the `netlify.toml` file that was just autogenerated as we'll need to add a few things to this. First, Netlify Dev needs to know more about your framework since it was not detected. Go ahead and add this to the toml:
```